src/sdk/object_io.js

```diff
@@ -467,7 +467,7 @@ class ObjectIO {
                     start: params.start,
                     end: params.start + chunk_info.size,
                     seq: params.seq,
-                    uncommitted: !params.complete_upload,
+                    uncommitted: typeof params.multipart_id === 'undefined' ? undefined : true,
```

Comment:
1. There is no need to use typeof x === 'undefined' in modern js, you can just use x === undefined. But actually in this we don't really care to identify the undefined vs other falsy values (which is the common case) so I would just use the Boolean() function for that.
2. It makes more sense to define it once outside of this chunks.map() call and define it one for all in the top level of this function.
3. It deserves a comment explaining how this uncommitted flag is used here.

```js
// on multipart upload we mark the parts as uncommitted 
// because we still need to fix their offsets on complete upload.
// otherwise, we immediately mark the parts as committed 
// because no further update is needed.
const uncommitted = Boolean(params.multipart_id);
```

Reply:
Ok, thanks.